### PR TITLE
Bump ginkgolinter to v0.9.0 and fix findings

### PR DESCRIPTION
**Pull request by nunnatsa**
**What this PR does / why we need it**:

Bump ginkgolinter to v0.9.0 and fix findings

The new version of ginkgo linter adds two new warnings:

* `HaveLen(0)` => `BeEmpty()`
* `Expect(x == y).To(BeTrue())` => `Expect(x).To(Equal(y))` or
`Expect(x > y).To(BeTrue())` => `Expect(x).To(BeNumerically(">", y))` etc.

```release-note
None
```

### CODE REVIEW

**Comment by iholder101** [comment]
Thanks @nunnatsa! Great stuff!
/lgtm

Just to be sure - the updated packages are direct dependencies for ginkgolinter, right?

**Comment by nunnatsa** [comment]
> Just to be sure - the updated packages are direct dependencies for ginkgolinter, right?

yes.

**Comment by nunnatsa** [comment]
Rebased (fixed conflicts)

**Comment by nunnatsa** [comment]
@iholder101 - please review again

**Review by EdDev — APPROVED** [review]

File: tests/libwait/wait.go [hunk 1]
```diff
@@ -92,10 +92,10 @@ func waitForVMIPhase(ctx context.Context, phases []v1.VirtualMachineInstancePhas
 		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, &metav1.GetOptions{})
 		gomega.ExpectWithOffset(1, err).ToNot(gomega.HaveOccurred())
 
-		gomega.Expect(vmi.Status.Phase == v1.Succeeded).To(gomega.BeFalse(), "VMI %s unexpectedly stopped. State: %s", vmi.Name, vmi.Status.Phase)
+		gomega.Expect(vmi.Status.Phase).ToNot(gomega.Equal(v1.Succeeded), "VMI %s unexpectedly stopped. State: %s", vmi.Name, vmi.Status.Phase)
```
Comment (xpivarc) [inline tests/libwait/wait.go:95]: Here we can use the custom matcher `BeInPhase`

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
Hold just in case you want to take the suggestion

File: tests/libwait/wait.go (hunk 1 above)
Comment (nunnatsa) [reply tests/libwait/wait.go]: Done

**Review by xpivarc — APPROVED** [review]

**Pull request merged**